frontend/packages/console-app/src/actions/creators/common-factory.ts

```diff
@@ -0,0 +1,116 @@
+import i18next from 'i18next';
+import { Action } from '@console/dynamic-plugin-sdk';
+import {
+  annotationsModal,
+  deleteModal,
+  labelsModal,
+  configureReplicaCountModal,
+  podSelectorModal,
+  tolerationsModal,
+} from '@console/internal/components/modals';
+import { resourceObjPath, asAccessReview } from '@console/internal/components/utils';
+import { referenceForModel, K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+
+export type ResourceActionCreator = (
+  kind: K8sKind,
+  obj: K8sResourceKind,
+  relatedResource?: K8sResourceKind,
+) => Action;
+
+export type ResourceActionFactory = { [name: string]: ResourceActionCreator };
+
+export const CommonActionFactory: ResourceActionFactory = {
+  Delete: (kind: K8sKind, obj: K8sResourceKind): Action => ({
+    id: `delete-resource`,
+    label: i18next.t('console-app~Delete {{kind}}', { kind: kind.kind }),
+    cta: () =>
+      deleteModal({
+        kind,
+        resource: obj,
+      }),
+    accessReview: asAccessReview(kind, obj, 'delete'),
+  }),
+  Edit: (kind: K8sKind, obj: K8sResourceKind): Action => ({
+    id: `edit-resource`,
+    label: i18next.t('console-app~Edit {{kind}}', { kind: kind.kind }),
+    cta: {
+      href: `${resourceObjPath(obj, kind.crd ? referenceForModel(kind) : kind.kind)}/yaml`,
+    },
+    // TODO: Fallback to "View YAML"? We might want a similar fallback for annotations, labels, etc.
+    accessReview: asAccessReview(kind, obj, 'update'),
+  }),
+  ModifyLabels: (kind: K8sKind, obj: K8sResourceKind): Action => ({
+    id: 'edit-labels',
+    label: i18next.t('console-app~Edit labels'),
+    cta: () =>
+      labelsModal({
+        kind,
+        resource: obj,
+        blocking: true,
+      }),
+    accessReview: asAccessReview(kind, obj, 'patch'),
+  }),
+  ModifyAnnotations: (kind: K8sKind, obj: K8sResourceKind): Action => ({
+    id: 'edit-annotations',
+    label: i18next.t('console-app~Edit annotations'),
+    cta: () =>
+      annotationsModal({
+        kind,
+        resource: obj,
+        blocking: true,
+      }),
+    accessReview: asAccessReview(kind, obj, 'patch'),
+  }),
+  ModifyCount: (kind: K8sKind, obj: K8sResourceKind): Action => ({
+    id: 'edit-pod-count',
+    label: i18next.t('console-app~Edit Pod count'),
+    cta: () =>
+      configureReplicaCountModal({
+        resourceKind: kind,
+        resource: obj,
+      }),
+    accessReview: asAccessReview(kind, obj, 'patch', 'scale'),
+  }),
```

Comment:
When testing from the deployment list page.

After editing the pod count and then reopening the modal, the number of pods is incorrect. It ends up showing the previous value still. 

Looks like there's an issue with the actions getting cached but not updated when the object changes.

Reply:
Updated. See comment above for the main issue.